src/main/scala/com/fulcrumgenomics/vcf/MakeTwoSampleMixtureVcf.scala

```diff
@@ -0,0 +1,200 @@
+          }
+          else {
+            builder.alleles(util.Arrays.asList(Allele.NO_CALL, Allele.NO_CALL))
+          }
+
+          (builder.make(), if (tumorOnly) None else Some(new GenotypeBuilder("normal", oldNormalGt.getAlleles).make()))
+        }
+
+        val gts = Seq(Some(newTumorGt), newNormalGt).flatten
+        val builder = new VariantContextBuilder(ctx.getSource, ctx.getContig, ctx.getStart(), ctx.getEnd(), ctx.getAlleles)
+        builder.genotypes(gts)
+
+        val filters = ListBuffer[String]()
+        if (isMultiAllelic) filters += MultiAllelicFilter
+        if (!tumorOnly && !oldNormalGt.isHomRef) filters += GermlineFilter
+        if (newTumorGt.isNoCall) filters += UnknownGtFilter
+        if (filters.nonEmpty) builder.filters(filters: _*) else builder.passFilters()
+        out.add(builder.make());
+      }
+    }
+
+    out.close()
+    in.safelyClose()
+  }
+
+  /** Gets the genotype of the sample, substituting in a HomRef genotype for no-calls if required. */
+  private def genotypeOf(sample: String, ctx: VariantContext): Genotype = {
+    val actual = ctx.getGenotype(sample)
+    if (actual.isNoCall && this.noCallIsHomRef) new GenotypeBuilder(sample, util.Arrays.asList(ctx.getReference)).make()
+    else actual
+  }
+
+  /** Makes a no-call genotype object with the provided sample name. */
+  private def noCall(sampleName: String) = {
+    new GenotypeBuilder(sampleName).alleles(util.Arrays.asList(Allele.NO_CALL, Allele.NO_CALL)).make()
+  }
+
+  /** Builds a header that can be used to write the mixture VCF. */
+  def makeWriter(in: VCFHeader, samples: String*): VariantContextWriter = {
+    val header = new VCFHeader(Collections.emptySet[VCFHeaderLine](), util.Arrays.asList(samples:_*))
+
+    in.getFilterLines.foreach(header.addMetaDataLine)
+    header.addMetaDataLine(new VCFFilterHeaderLine(GermlineFilter,  "Evidence seen in the normal sample"))
+    header.addMetaDataLine(new VCFFilterHeaderLine(UnknownGtFilter, "GT and AF not known due to one or more input samples being no-called."))
+    header.addMetaDataLine(new VCFFilterHeaderLine(MultiAllelicFilter, "Locus is multi-allelic in chosen samples."))
+
+    in.getFormatHeaderLines.foreach(header.addMetaDataLine)
+    header.addMetaDataLine(new VCFFormatHeaderLine("AF", 1, VCFHeaderLineType.Float, "Alt allele fraction in the tumor"))
+
+    in.getInfoHeaderLines.foreach(header.addMetaDataLine)
+    header.setSequenceDictionary(in.getSequenceDictionary)
+
+    val writer = new VariantContextWriterBuilder()
+      .setOutputFile(output.toFile)
+      .setOption(Options.INDEX_ON_THE_FLY)
+      .setReferenceDictionary(in.getSequenceDictionary)
+      .setOption(Options.ALLOW_MISSING_FIELDS_IN_HEADER)
+      .build()
+
+    writer.writeHeader(header)
+    writer
+  }
+
+  /** Generates an iterator over the whole file, or over the intervals if provided. */
+  def iterator(in: VCFFileReader, intervals: Option[PathToIntervals]): Iterator[VariantContext] = {
```

Comment:
this looks like something that would be nice in `FgbioDefs` as an implicit method on the VCFFileReader. What do you think?